doc/adr/0006-url-format-in-qr-code.md

```diff
@@ -0,0 +1,49 @@
+# 6. URL format in QR code
+
+Date: 2022-06-24
+
+## Status
+
+Accepted
+
+## Context
+
+The web widget displays a QR code, which transfers data to the mobile device, i.e. into the eID client.
+
+The QR code needs to include the ClientURL, including the `tcTokenURL`, (see [TR-03124 part 1](https://www.bsi.bund.de/SharedDocs/Downloads/DE/BSI/Publikationen/TechnischeRichtlinien/TR03124/TR-03124-1.pdf) for more information).
+
+Additionally, some kind of widget session id needs to be added, so that the proxy message server can identify this exact widget when the mobile app wants to send a message to the widget.
+
+Lastly, an [encryption key](https://github.com/digitalservicebund/useid-architecture/blob/df7596d0725fdbfab70f08d3b29a5641ac711fde/doc/adr/0004-e2ee-for-refresh-address.md) needs to added. This should never be sent to any server and only be known to the widget and the mobile app.
+
+On mobile devices, the user won't see a QR code, but only a link. The app behaves differently on mobile devices (opening the refresh address instead of informing the UseID Backend Server), so the mobile app needs to know if the user started from mobile device or not.
+
+## Decision
+
+The URL format follows this schema:
+
+```
+eid://127.0.0.1:24727/eID-Client?tcTokenURL={{tcTokenURL}}&widgetSessionId={{widgetSessionId}}&mobile={{isMobileDevice}}#{{encryptionKey}}
+```
+
+The encryption key (`{{encryptionKey}}`) is base64 encoded.
+
+The mobile flag (`{{isMobileDevice}}`) has boolean values (`true` or `false`).
+
+## Consequences
+
+The URL follows the Technical Guideline and is compatible to other eID clients.
+
+The URL contains all the needed data, including the `widgetSessionId` and the encryption key.
+
+The encryption key is not sent to any server, since it's added as fragment id.
```

Comment:
The ClientURL is never sent to any server. It is just encoded in the QR code and read by the mobile app. Why shouldn't the encryption key be included as a regular query parameter?

Reply:
Good question. This whole URL makes actually no sense at all. But since it now is an URL, we can be double sure for any future changes by using fragment id. I'll add that.